src/NHibernate/Impl/AbstractSessionImpl.cs

```diff
@@ -380,31 +424,31 @@ public virtual ISQLQuery CreateSQLQuery(string sql)
 
 		protected internal virtual IQueryExpressionPlan GetHQLQueryPlan(IQueryExpression queryExpression, bool shallow)
 		{
-			using (new SessionIdLoggingContext(SessionId))
+			using (BeginContext())
 			{
 				return _factory.QueryPlanCache.GetHQLQueryPlan(queryExpression, shallow, EnabledFilters);
```

Comment:
For this case I was tempted to put a `BeginProcess`, because it does not really make sens to get the query plan on a closed session. But for sticking to previous semantic, and since it apparently calls nothing that does the session status check or requires it to be opened, I left it as is.
But checking again, `EnabledFilters` does check the session status in the `SessionImpl` override, so I am now switching it to `BeginProcess`.